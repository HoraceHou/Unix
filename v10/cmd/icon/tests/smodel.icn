procedure main()
   write(image(t := table()) | "failed")
   write(image(every i := 1 to 20 do t[i] := i) | "failed")
   write(image(?t) | "failed")
   write(image(?t) | "failed")
   write(image(?t) | "failed")
   write(image(?t) | "failed")
   write(image(?t) | "failed")
   write(image(?t) | "failed")
   write(image(?t) | "failed")
   write(image(?t) | "failed")
   write(image(?t) | "failed")
   write(image(?t) | "failed")
   write(image(t := table()) | "failed")
   write(image(every t[c := !(12345678 || &lcase)] := c) | "failed")
   write(image(every write(!t)) | "failed")
   write(image(set([1,0,1,0,1,0,1,0])) | "failed")
   write(image(set([])) | "failed")
   write(image(s := set([1,2,3,4,5,6,7,8,9,10])) | "failed")
   write(image(?s) | "failed")
   write(image(?s) | "failed")
   write(image(every write(!s)) | "failed")
   write(image(s := set([])) | "failed")
   write(image(every insert(s,1 to 20)) | "failed")
   write(image(every write(!s)) | "failed")
   write(image(every delete(s,15 to 30)) | "failed")
   write(image(every write(!s)) | "failed")
   write(image(every s1 := insert(set([]),!&lcase)) | "failed")
   write(image(s2 := set(["a","aa","ab","b",1,2,3,4])) | "failed")
   write(image(s3 := s1 ++ s2) | "failed")
   write(image(s4 := s1 ** s2) | "failed")
   write(image(s5 := s1 -- s2) | "failed")
   write(image(*s3) | "failed")
   write(image(every write(!s3)) | "failed")
   write(image(*s4) | "failed")
   write(image(every write(!s4)) | "failed")
   write(image(every write(!s5)) | "failed")
end
