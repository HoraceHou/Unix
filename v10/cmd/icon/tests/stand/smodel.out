table(0)
failed
15
17
16
18
17
16
1
20
1
20
table(0)
failed
3
g
t
4
h
u
5
i
v
6
j
w
7
k
x
8
l
y
m
z
a
n
b
o
c
p
d
q
1
e
r
2
f
s
failed
set(2)
set(0)
set(10)
2
4
1
2
3
4
5
6
7
8
9
10
failed
set(0)
failed
13
1
14
2
15
3
16
4
17
5
18
6
19
7
20
8
9
10
11
12
failed
failed
13
1
14
2
3
4
5
6
7
8
9
10
11
12
failed
failed
set(8)
set(32)
set(2)
set(24)
32
g
t
1
h
u
aa
2
i
v
ab
3
j
w
4
k
x
l
y
m
z
a
n
b
o
c
p
d
q
e
r
f
s
failed
2
a
b
failed
g
t
h
u
i
v
j
w
k
x
l
y
m
z
n
o
c
p
d
q
e
r
f
s
failed
